tests/loggen/loggen.c

```diff
@@ -21,956 +20,400 @@
-  { "loop-reading",'l',0, G_OPTION_ARG_NONE, &loop_reading, "Read the file specified in read-file option in loop (it will restart the reading if reached the end of the file)", NULL },
-  { "dont-parse", 'd', 0, G_OPTION_ARG_NONE, &dont_parse, "Don't parse the lines coming from the readed files. Loggen will send the whole lines as it is in the readed file", NULL },
-  { "skip-tokens", 0, 0, G_OPTION_ARG_INT, &skip_tokens, "Skip the given number of tokens (delimined by a space) at the beginning of each line (default value: 3)", "<number>" },
-  { NULL }
-};
-
-void
-version(void)
+static void
+signal_callback_handler(int signum)
 {
-  printf(SYSLOG_NG_PACKAGE_NAME " " SYSLOG_NG_VERSION "\n");
+  ERROR("Send error Broken pipe, results may be skewed. %d\n",signum);
 }
 
+
 int
 main(int argc, char *argv[])
 {
-  int ret = 0;
-  GError *error = NULL;
-  GOptionContext *ctx = NULL;
-  int i;
-  guint64 diff_usec;
-  GOptionGroup *group;
+  GPtrArray *plugin_array = g_ptr_array_new();
+  GOptionContext *ctx = g_option_context_new(" target port");
 
-  g_thread_init(NULL);
-  tzset();
+  signal(SIGPIPE, signal_callback_handler);
 
-  signal(SIGPIPE, SIG_IGN);
+  int plugin_num = enumerate_plugins(SYSLOG_NG_PATH_LOGGEN_PLUGIN_DIR, plugin_array, ctx);
+  DEBUG("%d plugin successfuly loaded\n",plugin_num);
 
-  ctx = g_option_context_new(" target port");
-  g_option_context_add_main_entries(ctx, loggen_options, 0);
-
-
-  group = g_option_group_new("file", "File options", "Show file options", NULL, NULL);
-  g_option_group_add_entries(group, file_option_entries);
+  // create sub group for file reader functions
+  GOptionGroup *group = g_option_group_new("File reader", "File reader", "Show options", NULL, NULL);
+  g_option_group_add_entries(group, get_file_reader_options());
   g_option_context_add_group(ctx, group);
 
+  GError *error = NULL;
   if (!g_option_context_parse(ctx, &argc, &argv, &error))
     {
-      fprintf(stderr, "Option parsing failed: %s\n", error->message);
+      ERROR("option parsing failed: %s\n", error->message);
+      g_ptr_array_free(plugin_array,TRUE);
       return 1;
     }
 
-  if (display_version)
+  if (argc<3)
     {
-      version();
-      return 0;
-    }
-
-  if (active_connections <= 0)
-    {
-      fprintf(stderr, "Minimum value of active-connections must be greater than 0\n");
+      ERROR("please specify target and port at least\n");
+      g_ptr_array_free(plugin_array,TRUE);
       return 1;
     }
 
-  if (unix_socket_i)
-    unix_socket = 0;
-  if (unix_socket_x)
-    unix_socket = 1;
-
-  if (sock_type_d)
-    sock_type = SOCK_DGRAM;
-  if (sock_type_s)
-    sock_type = SOCK_STREAM;
-
-  if (message_length > MAX_MESSAGE_LENGTH)
-    {
-      fprintf(stderr, "Message size too large, limiting to %d\n", MAX_MESSAGE_LENGTH);
-      message_length = MAX_MESSAGE_LENGTH;
-    }
+  global_plugin_option.target = g_strdup(argv[1]);
```

Comment:
These should be free-ed due to g_strdup.

Reply:
done